### PR TITLE
fix/adata: fix indices checking for permissions and owner append functions in AData

### LINKED ISSUE #880: Add index fields to AddADataPermissions and SetDataOwner
As mentioned maidsafe/rfcs#318. 

The `AddADataPermissions` field should have a `permissions_index` which is validated before a new permission set is appended. The `permissions_index` should hold the last index of the permissions field.

Similarly for `SetADataOwners` an `owners_index` field should be added to the request. This is validated at the vaults.

The changes to the RFC were merged in maidsafe/rfcs#328

### PR DESCRIPTION
**Pull request by Yoga07**
Resolves #880

### CODE REVIEW

File: safe_core/src/client/mock/vault.rs [hunk 1]
```diff
@@ -934,13 +916,13 @@ impl Vault {
                     .get_adata(append.address, requester_pk, request)
                     .and_then(move |data| match data {
                         AData::PubSeq(mut adata) => {
-                            unwrap!(adata.append(&append.values, index));
+                            unwrap!(adata.append(append.values, index));
```
Comment (lionel1704) [inline safe_core/src/client/mock/vault.rs]: Can you make this change please?
```suggestion
                            adata.append(append.values, index)?;
```
`?` will return the error into the result which will be returned in the response.

File: safe_core/src/client/mock/vault.rs [hunk 2]
```diff
@@ -934,13 +916,13 @@ impl Vault {
...
                             self.commit_mutation(requester.name());
                             self.insert_data(id, Data::AppendOnly(AData::PubSeq(adata)));
                             Ok(())
                         }
                         AData::UnpubSeq(mut adata) => {
-                            unwrap!(adata.append(&append.values, index));
+                            unwrap!(adata.append(append.values, index));
```
Comment (lionel1704) [inline safe_core/src/client/mock/vault.rs]: Here too.

File: safe_core/src/client/mock/vault.rs [hunk 3]
```diff
@@ -955,13 +937,13 @@ impl Vault {
                     .get_adata(append.address, requester_pk, request)
                     .and_then(move |data| match data {
                         AData::PubUnseq(mut adata) => {
-                            unwrap!(adata.append(&append.values));
+                            unwrap!(adata.append(append.values));
```
Comment (lionel1704) [inline safe_core/src/client/mock/vault.rs]: Here and a few more places where `append` and `append_permissions` is called

**Comment by Yoga07** [comment]
Closing this PR as the changes(along with it's comments) are being handled in PR maidsafe/safe_client_libs#891  .

**Pull request closed without merging**